### PR TITLE
Fixes for Ruby 2.4 and broken tests

### PR DESCRIPTION
**Pull request by mdpye**
- don't use the `Fixnum` type (deprecated in Ruby 2.4)
- upgrade deps and pin versions where necessary
- fix argument ordering problem which was evident from failing tests

### CODE REVIEW

File: lib/em-hiredis/connection_manager.rb [hunk 1]
```diff
@@ -44,12 +44,12 @@ class ConnectionManager
...
     #   of EMConnection or fails if the connection was unsuccessful.
     #   Failures will be retried
-    def initialize(connection_factory, em = EM, reconnect_attempts)
+    def initialize(connection_factory:, reconnect_attempts: nil, em: EM)
 
-      @reconnect_attempts = initialise_reconnect_attempts(reconnect_attempts)
       @em = em
       @connection_factory = connection_factory
 
+      @reconnect_attempts = reconnect_attempts || DEFAULT_RECONNECT_ATTEMPTS
```
Comment (leesio) [inline lib/em-hiredis/connection_manager.rb:52]: I guess we can get rid of the `initialise_reconnect_attempts` method too.

**Review by leesio — APPROVED** [review]
Looks good. Just a small comment

**Comment by mdpye** [comment]
@leesio I probably should have fixed the travis test run as well as the local one! Better now, after who knows how long of failing because there was no linked redis to test against :/

**Comment by leesio** [comment]
Indeed.

The hipchat integration is probably a good indicator of how recently it's been updated.

**Review by leesio — APPROVED** [review]

**Pull request merged**